stylus-sdk/src/storage/bytes.rs

```diff
@@ -275,31 +271,181 @@ impl StorageBytes {
 
 impl Erase for StorageBytes {
     fn erase(&mut self) {
-        let mut len = self.len() as isize;
+        let root = BytesRoot::new(self);
+        let mut len = root.len() as isize;
+        // clear any slots used in long storage
         if len > 31 {
             while len > 0 {
-                let slot = self.index_slot(len as usize - 1).0;
-                unsafe { Storage::clear_word(self.__stylus_host.clone(), slot) };
+                let slot = root.index_slot(len as usize - 1).0;
+                unsafe { Storage::clear_word(root.storage.clone_vm(), slot) };
                 len -= 32;
             }
         }
-        unsafe { Storage::clear_word(self.__stylus_host.clone(), self.root) };
+        // set length and data in root storage to zero
+        unsafe { Storage::clear_word(root.storage.clone_vm(), root.storage.root) };
     }
 }
 
 impl Extend<u8> for StorageBytes {
     fn extend<T: IntoIterator<Item = u8>>(&mut self, iter: T) {
-        for elem in iter {
-            self.push(elem);
+        let mut iter = iter.into_iter();
+        let mut root = BytesRoot::new_mut(self);
+        let mut len = root.len();
+
+        // if storage is small, grow it until it reaches 32 bytes
+        if len < 32 {
+            while len < 32 {
+                if let Some(byte) = iter.next() {
+                    root.word[len] = byte;
```

Comment:
Do we need a bounds check here?

Reply:
It is not necessary because the `while` condition already ensures that len is smaller than the word size.